### ISSUE TITLE
[CI/CD]: Review coverage of accessibility checks in GI Bill Comparison Tool end-to-end tests

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#7373]
**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Applications **must** have thorough end-to-end tests that run in our continuous integration/continuous deployment (CI/CD) pipeline. These tests should include thorough axe checks. While auditing the GI Bill Comparison Tool e2e tests, I noticed a few things worth a second look:

* The provider detail view is showing an error page, not an actual education provider. I tried this locally with mock data, with and without the `vets-api` running.
* The provider detail view should expand the `Edit eligibility details` accordion before running an axe check.
* The provider detail should open at least one `Learn more` modal window and run an axe check.

I'd like the front-end team to update their e2e axe checks to include a branch that tests the first page with this dropdown present. Definition of done in acceptance criteria below.

## Related Issues
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/7727

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Environment
* `vets-website/src/applications/gi/tests`
* `$ yarn test:e2e src/applications/gi/tests`

## Acceptance Criteria

**Definition of done:**
- [ ] Front-end team member(s) have reviewed end-to-end tests for axe checks
- [ ] Ensure provider view e2e test resolves to an actual provider
- [ ] Expand the `Edit eligibility details` accordion on the provider page, then run the axe check
- [ ] Open an modal window on the provider detail view, and run another axe check
- [ ] If possible, maintain the existing error page for provider, and run that axe check
- [ ] FE team has consulted with accessibility specialist in cases where there are high numbers of modals, accordions, other hidden content that could slow down e2e test runs.
- [ ] No axe `violations` appear in the plugin console. These will break the CI/CD build.

## WCAG or Vendor Guidance (optional)

* [Custom axeCheck helper method](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/testing/e2e/nightwatch-commands/axeCheck.js)
* [VSP guidance on writing end-to-end tests](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/platform/quality-assurance/e2e-testing)

**Comment by 1Copenut** [comment]
@desireeturner && @mattpointzxer0 
I'm going to escalate this issue because I found an axe error in our full provider view. e2e tests with axe checks will serve as an early warning if we get a regression in the future. I'll cross-link the new issue after lunch.

**Comment by zurbergram** [comment]
https://github.com/department-of-veterans-affairs/vets-website/pull/12190/

**Comment by desireeturner** [comment]
This issue is blocked by #7272 and #7280

**Comment by zurbergram** [comment]
@1Copenut This is ready for review

**Comment by 1Copenut** [comment]
@zurbergram This is on point! I took a look at the e2e tests locally, ran them a few times and saw the axe checks inside your helpers. I'll move this to close.
